capellambse/model/__init__.py

```diff
@@ -324,3 +328,171 @@ def search(
+            executable and an accessible Docker image in parallel. In
+            such a case the caller can force to get the value for
+            *capella_cli* interpreted as an executable with *force* set
+            to ``"exe"`` or as a Docker image when *force* is set to
+            ``"img"`` (Docker image). If *force* is not specified and an
+            ambiguous name is specified for *capella_cli*, the local
+            executable will be used and not the accessible Docker
+            image with the same name.
+
+        Raises
+        ------
+        TypeError
+            If there has no diagram cache being specified during load of
+            the model.
+
+        Examples
+        --------
+        **Running a local installation of Capella**
+
+        All the following examples call the method
+        :meth:`update_diagram_cache` on a model
+        for which it is assumed that it has been loaded like that:
+
+        >>> model = MelodyModel(
+        ...    "/path/to/model.aird",
+        ...    diagram_cache="/path/to/diagram_cache"
+        ... )
+
+        Passing an executable/ symlink named ``capella`` that is in the
+        ``PATH`` environment variable:
+
+        >>> model.update_diagram_cache(
+        ...     "capella", "png", True
+        ... )
+
+        Passing an absolute path to a local installation of Capella that
+        contains the Capella version:
+
+        .. note::
+            Here, the algorithm will replace the placeholder
+            ``{VERSION}`` by the x.y.z. formatted version to which the
+            given model has been migrated. Therefore, for a model that
+            has been migrated to Capella 6.0, it will be tried to run
+            the executable
+            ``~/.bin/Capella_6.0.0.app/Contents/MacOS/capella``
+
+        >>> model.update_diagram_cache(
+        ...     "~/.bin/Capella_{VERSION}.app/Contents/MacOS/capella"
+        ... )
+        """
+        # TODO: More examples in docstring!
+        # pylint: enable=line-too-long
+        if not hasattr(self, "_diagram_cache"):
+            raise TypeError(
+                "There is no diagram cache specified that can be updated. "
+                "To use this a diagram cache must be specified when a model "
+                "is loaded."
+            )
+        diag_cache_dir = pathlib.Path(self._diagram_cache.path)
+        if not diag_cache_dir.is_dir():
+            raise ValueError(
+                f"The specified diagram_cache location '{diag_cache_dir}' is "
+                "invalid. Configure an existing directory, please!"
+            )
+        else:
+            # empty cache:
+            shutil.rmtree(diag_cache_dir)
+            diag_cache_dir.mkdir()
+        capella_version = self.info.capella_version
+        assert (
+            capella_version is not None
+        ), "'ModelInfo.capella_version' is None!"
+        assert re.match(r"\d+\.\d+\.\d+", capella_version) is not None, (
```

Comment:
Redundant `is not None`, `Match` objects are always truthy.

Reply:
Okay, thx. How would you write the check, if `capella_version` matches the given pattern?

Reply:
I only meant that you don't need to write `assert re.match(...) is not None`, and `assert re.match(...)` is already enough. I'm fine with simply having this as assertion, because it holds true for all Capella versions that we currently support anyways. :)